lib/features/qr_code_scan/presenter/pages/qr_code/qr_code_page.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:superformula_leandro/core/constants/strings_constants.dart';
+import 'package:superformula_leandro/core/widgets/app_primary_button.dart';
+import 'package:superformula_leandro/features/qr_code_scan/presenter/cubits/qr_code/qr_code_cubit.dart';
+import 'package:superformula_leandro/features/qr_code_scan/presenter/cubits/timer/timer_cubit.dart';
+import 'package:superformula_leandro/core/widgets/app_error_widget.dart';
+import 'package:superformula_leandro/features/qr_code_scan/presenter/pages/qr_code/widgets/qr_code_loaded_body_widget.dart';
+import 'package:superformula_leandro/features/qr_code_scan/presenter/pages/qr_code/widgets/qr_code_loading_body_widget.dart';
+
+class QrCodePage extends StatefulWidget {
+  const QrCodePage({super.key});
+
+  @override
+  State<QrCodePage> createState() => _QrCodePageState();
+}
+
+class _QrCodePageState extends State<QrCodePage> {
+  late final TimerCubit _timerCubit;
+  late final QrCodeCubit _qrCodeCubit;
+
+  @override
+  void initState() {
+    super.initState();
+    _qrCodeCubit = context.read<QrCodeCubit>()..getQrCode();
+    _timerCubit = context.read<TimerCubit>();
+  }
+
+  @override
+  void dispose() {
+    _timerCubit.disposeTimer();
```

Comment:
thought: great catch disposing the timer if disposing the screen, butyou can also dispose the timer on the cubit's dispose method.

Reply:
Yeah, of course. Thanks.